app/src/main/java/com/trempel/ui/main_activity/MainActivity.kt

```diff
@@ -48,10 +55,58 @@ internal class MainActivity : AppCompatActivity() {
             containerId = R.id.container_view,
             null
         )
+        val appBarConfiguration = AppBarConfiguration(
+            topLevelDestinationIds = setOf(
+                R.id.homePageFragment,
+                R.id.categoryFragment,
+                R.id.bagFragment,
+                R.id.favouritesFragment,
+                R.id.profileFragment
+            ),
+            fallbackOnNavigateUpListener = ::onSupportNavigateUp
+        )
+        controller.observe(this) {
+            binding.toolbar.setupWithNavController(it, appBarConfiguration)
+        }
         currentNavController = controller
     }
 
     override fun onSupportNavigateUp(): Boolean {
         return currentNavController?.value?.navigateUp() ?: false
     }
+
+    private fun ActivityMainBinding.backPressedListener() {
```

Comment:
since we have `binding` as a field now, we can transform this extension function to regular one